apps/cyberstorm-nextjs/app/communities/layout.tsx

```diff
@@ -0,0 +1,19 @@
+import "@thunderstore/cyberstorm-styles";
+import rootStyles from "../RootLayout.module.css";
+import { BreadCrumbs } from "@thunderstore/cyberstorm";
+import React from "react";
+import { PageHeader } from "@thunderstore/cyberstorm/src/components/Layout/BaseLayout/PageHeader/PageHeader";
+
+export default function CommunitiesLayout(props: React.PropsWithChildren) {
```

Comment:
I gave a quick glance to [the documentation](https://nextjs.org/docs/pages/building-your-application/routing/pages-and-layouts#layout-pattern) and it seems to me the approach there is a bit different from the approach used here. How much though have you given to this, to ensure we have as DRY implementation as possible while still retaining necessary flexibility? (This is an unloaded question, honestly IDK how this should be done.)

Reply:
To have as DRY implementation as possible, we'll have to change designs, which is not something I want to get into right now.

Currently I've just focused on getting everything moved under NextJS and after that figure out the best structure and separation.

So in practice the implementation/convertion will be something like this:
BaseLayout -> RootLayout
Cyberstorm Layouts -> Route specific layouts
Components/"Sub-cyberstorm-layouts" inside Cyberstorm layouts -> NextJS Slots or Pages. Depending on the case.

With this setup we should not have too much repetition and still have all the flexibility we need.

TLDR; Have thought about this. Don't have a clear answer. Best to go forward and figure it out as we go 👍